e2etest/bgptests/node_selector.go

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"context"
+	"fmt"
+
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	"go.universe.tf/metallb/e2etest/pkg/k8s"
+	"go.universe.tf/metallb/e2etest/pkg/metallb"
+	testservice "go.universe.tf/metallb/e2etest/pkg/service"
+	metallbconfig "go.universe.tf/metallb/internal/config"
+	"go.universe.tf/metallb/internal/ipfamily"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+
+	"go.universe.tf/metallb/e2etest/pkg/frr"
+	frrconfig "go.universe.tf/metallb/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/metallb/e2etest/pkg/frr/container"
+	corev1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = ginkgo.Describe("BGP Node Selector", func() {
+	var cs clientset.Interface
+	var f *framework.Framework
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
+			for _, c := range FRRContainers {
+				address := c.Ipv4
+				if address == "" {
+					address = c.Ipv6
+				}
+				peerAddr := address + fmt.Sprintf(":%d", c.RouterConfig.BGPPort)
+				dump, err := frr.RawDump(c, "/etc/frr/bgpd.conf", "/tmp/frr.log", "/etc/frr/daemons")
+				framework.Logf("External frr dump for %s:%s\n%s\nerrors:%v", c.Name, peerAddr, dump, err)
+			}
+
+			speakerPods, err := metallb.SpeakerPods(cs)
+			framework.ExpectNoError(err)
+			for _, pod := range speakerPods {
+				if len(pod.Spec.Containers) == 1 { // we dump only in case of frr
+					break
+				}
+				podExec := executor.ForPod(pod.Namespace, pod.Name, "frr")
+				dump, err := frr.RawDump(podExec, "/etc/frr/frr.conf", "/etc/frr/frr.log")
+				framework.Logf("External frr dump for pod %s\n%s %v", pod.Name, dump, err)
+			}
+			k8s.DescribeSvc(f.Namespace.Name)
+		}
+	})
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Clearing any previous configuration")
+
+		err := ConfigUpdater.Clean()
+		framework.ExpectNoError(err)
+
+		for _, c := range FRRContainers {
+			err := c.UpdateBGPConfigFile(frrconfig.Empty)
+			framework.ExpectNoError(err)
+		}
+	})
+
+	f = framework.NewDefaultFramework("bgp")
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+	})
+
+	table.DescribeTable("Two services, two distinct advertisements with different node selectors",
+		func(pairingIPFamily ipfamily.Family, addresses []string, nodesForFirstPool, nodesForSecondPool []int) {
+			var allNodes *corev1.NodeList
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			framework.ExpectNoError(err)
+
+			expectedNodesForFirstPool := nodesForSelection(allNodes.Items, nodesForFirstPool)
```

Comment:
can we use `"k8s.io/kubernetes/test/e2e/framework/node"`'s GetBoundedReadySchedulableNodes func instead? (we should probably leverage that package more anyways)

also, couldn't we get the same nodes for first/second pool (do we care about it here)?